File: Backend/MATCHOP.API/Entities/Venue.cs
using MATCHOP.API.Enums;

namespace MATCHOP.API.Entities;

public class Venue
{
    public Guid Id { get; set; }
    public Guid OwnerId { get; set; }
    public string Name { get; set; } = null!;
    public string Address { get; set; } = null!;
    public string City { get; set; } = null!;
    public string District { get; set; } = null!;
    public string? Ward { get; set; }
    public decimal? Latitude { get; set; }
    public decimal? Longitude { get; set; }
    public string? Description { get; set; }
    public TimeOnly OpeningTime { get; set; }
    public string? CoverImageUrl { get; set; }
    public TimeOnly ClosingTime { get; set; }
    public VenueStatus Status { get; set; } = VenueStatus.DRAFT;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User Owner { get; set; } = null!;
    public ICollection<Court> Courts { get; set; } = new List<Court>();
    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();
    public ICollection<CourtBlock> CourtBlocks { get; set; } = new List<CourtBlock>();
}

File: Frontend/src/types/venue.ts
export type VenueStatus =
  | "DRAFT"
  | "PENDING_APPROVAL"
  | "ACTIVE"
  | "INACTIVE"
  | "SUSPENDED"
  | "REJECTED";

export interface Venue {
  id: string;
  ownerId: string;
  name: string;
  address: string;
  city: string;
  district: string;
  ward?: string;
  latitude?: number;
  longitude?: number;
  description?: string;
  openingTime: string;
  closingTime: string;
  coverImageUrl?: string;
  status: VenueStatus;
  createdAt: string;
  updatedAt: string;
  courts?: Court[];
}

export interface CreateVenueRequest {
  name: string;
  address: string;
  city: string;
  district: string;
  ward?: string;
  description?: string;
  openingTime: string;
  closingTime: string;
  coverImage?: File;
}

export interface UpdateVenueRequest extends Partial<CreateVenueRequest> {}

import { Court } from "./court";
